source/draw/DrawState.ooc

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use geometry
+import Pen, Image, Map
+
+// Example
+// DrawState new(myImage) setMap(myShader) draw()
+
+DrawState: cover {
+	target: Image = null
+	inputImage: Image = null
+	map: Map = null
+	opacity := 1.0f
+	_transformNormalized := FloatTransform3D identity
+	viewportLocal := IntBox2D new(0, 0, 0, 0)
+	init: func@ ~default
+	init: func@ ~target (=target)
+	setTarget: func (target: Image) -> This {
+		result := this
+		result target = target
+		result
+	}
+	setMap: func (map: Map) -> This {
+		result := this
+		result map = map
+		result
+	}
+	setOpacity: func (opacity: Float) -> This {
+		result := this
+		result opacity = opacity
+		result
+	}
+	setViewport: func (viewport: IntBox2D) -> This {
+		result := this
+		result viewportLocal = viewport
+		result
+	}
+	setInputImage: func (inputImage: Image) -> This {
+		result := this
+		result inputImage = inputImage
+		result
+	}
+	_toNormalized: static func (imageSize: FloatVector2D, transform: FloatTransform3D) -> FloatTransform3D {
```

Comment:
Is there a place for these conversion functions that I have everywhere in my code?